cmd/dockerd/daemon.go

```diff
@@ -287,7 +288,9 @@ func (cli *DaemonCli) start(opts daemonOptions) (err error) {
 	cli.d = d
 
 	// initMiddlewares needs cli.d to be populated. Dont change this init order.
-	cli.initMiddlewares(api, serverConfig)
+	if err = cli.initMiddlewares(api, serverConfig); err != nil {
```

Comment:
Is the `err =` on purpose? I find it dangerous, knowing that the function returns a named value. I would prefer `err :=`